src/rust/engine/src/rule_graph.rs

```diff
@@ -792,13 +792,13 @@ pub fn params_str(params: &ParamTypes) -> String {
 
 pub fn select_key_str(select_key: &SelectKey) -> String {
   match select_key {
-    &SelectKey::JustSelect(ref s) => select_str(s),
+    &SelectKey::JustSelect(ref s) => s.product.to_string(),
     &SelectKey::JustGet(ref g) => get_str(g),
   }
 }
 
-pub fn select_str(select: &Select) -> String {
-  format!("Select({})", select.product).to_string() // TODO variant key
+pub fn select_root_str(select: &Select) -> String {
+  format!("Select({})", select.product)
```

Comment:
Should we rename `Select` to `Root` or `SelectRoot` or something here, now that that's what it represents?

Reply:
"It's complicated", because `Select` is still the `Node` that we run to kick off a request for the parameters to a rule that is going to run, and in that case it isn't a root.